tools/mtmd/clip.cpp

```diff
@@ -3634,9 +4094,46 @@ bool clip_image_preprocess(struct clip_ctx * ctx, const clip_image_u8 * img, str
         res_imgs->grid_y = inst.grid_size.height;
         return true;
 
-    } else if ( ctx->proj_type() == PROJECTOR_TYPE_LFM2
-             || ctx->proj_type() == PROJECTOR_TYPE_KIMIVL
-    ) {
+    } else if (ctx->proj_type() == PROJECTOR_TYPE_JINACLIP) {
+        // JinaCLIP-specific preprocessing: Resize shortest edge to target, then center-crop to square
+        // to match Python reference (PIL Resize(shortest) + CenterCrop)
+        clip_image_u8 processed_image;
+        int           sz = params.image_size;  // e.g., 512 for JinaCLIP
+
+        // 1) Preserve aspect ratio: resize so that the shorter side == sz (bicubic)
+        int in_w = img->nx;
+        int in_h = img->ny;
+        int out_w, out_h;
+        if (in_w <= 0 || in_h <= 0) {
+            LOG_ERR("%s: invalid input image size %dx%d\n", __func__, in_w, in_h);
+            return false;
+        }
+        if (in_w < in_h) {
+            out_w = sz;
+            out_h = std::max(1, (int) std::round((double) in_h * sz / in_w));
+        } else {
+            out_h = sz;
+            out_w = std::max(1, (int) std::round((double) in_w * sz / in_h));
+        }
+
+        clip_image_u8 resized_keep_ratio;
+        image_manipulation::bicubic_pil_resize(*img, resized_keep_ratio, out_w, out_h);
```

Comment:
Generally pre-processing doesn't need to be byte-exact. I would prefer keeping the old `bicubic_resize` to keep it simple.